File: cinema-admin-frontend/src/app/repertory/dtos/create-screening-request.ts
import {MovieVariantResponse} from "../../movie/dtos/response/movie-variant.response";

export class CreateScreeningRequest {
  constructor(
    readonly movieTitle: string,
    readonly startTime: Date,
    readonly endTime: Date,
    readonly screeningRoom: string,
    readonly email: string,
    readonly movieVariant: MovieVariantResponse
  ) {
  }
}

File: cinema-backend/src/main/java/org/example/cinemabackend/cinema/application/dto/request/create/CreateScreeningRequest.java
package org.example.cinemabackend.cinema.application.dto.request.create;

import lombok.Builder;
import org.example.cinemabackend.movie.application.dto.response.MovieVariantResponse;

import java.time.LocalDateTime;

@Builder
public record CreateScreeningRequest(
        String movieTitle,
        LocalDateTime startTime,
        LocalDateTime endTime,
        String screeningRoom,
        MovieVariantResponse movieVariant
) {
}
